crates/api_models/src/payments.rs

```diff
@@ -1189,6 +1189,17 @@ impl std::fmt::Display for PaymentIdType {
     }
 }
 
+impl PaymentIdType {
+    pub fn get_id(&self) -> String {
+        match &self {
+            Self::PaymentIntentId(a)
+            | Self::ConnectorTransactionId(a)
+            | Self::PaymentAttemptId(a)
+            | Self::PreprocessingId(a) => a.to_owned(),
```

Comment:
Why only PreprocessingId is owned?

Reply:
No it is owned for all the above values as well